Libraries/WebSocket/WebSocket.js

```diff
@@ -72,53 +121,53 @@ class WebSocket extends WebSocketBase {
   }
 
   _unregisterEvents(): void {
-    this._subs.forEach(e => e.remove());
-    this._subs = [];
+    this._subscriptions.forEach(e => e.remove());
+    this._subscriptions = [];
   }
 
-  _registerEvents(id: number): void {
-    this._subs = [
+  _registerEvents(): void {
+    this._subscriptions = [
       RCTDeviceEventEmitter.addListener('websocketMessage', ev => {
-        if (ev.id !== id) {
+        if (ev.id !== this._socketId) {
           return;
         }
         var event = new WebSocketEvent('message', {
           data: (ev.type === 'binary') ? base64.toByteArray(ev.data).buffer : ev.data
         });
-        this.onmessage && this.onmessage(event);
         this.dispatchEvent(event);
       }),
       RCTDeviceEventEmitter.addListener('websocketOpen', ev => {
-        if (ev.id !== id) {
+        if (ev.id !== this._socketId) {
           return;
         }
         this.readyState = this.OPEN;
         var event = new WebSocketEvent('open');
-        this.onopen && this.onopen(event);
         this.dispatchEvent(event);
       }),
       RCTDeviceEventEmitter.addListener('websocketClosed', ev => {
-        if (ev.id !== id) {
+        if (ev.id !== this._socketId) {
           return;
         }
         this.readyState = this.CLOSED;
         var event = new WebSocketEvent('close');
         event.code = ev.code;
         event.reason = ev.reason;
-        this.onclose && this.onclose(event);
         this.dispatchEvent(event);
         this._unregisterEvents();
         this.close();
       }),
       RCTDeviceEventEmitter.addListener('websocketFailed', ev => {
-        if (ev.id !== id) {
+        if (ev.id !== this._socketId) {
           return;
         }
         var event = new WebSocketEvent('error');
         event.message = ev.message;
-        this.onerror && this.onerror(event);
-        this.onclose && this.onclose(event);
         this.dispatchEvent(event);
+
+        event = new WebSocketEvent('close');
+        event.message = ev.message;
+        this.dispatchEvent(event);
+
```

Comment:
Seems like we're in agreement then. :)

I think I can count myself as one of React Native's authors, so I'll just leave them in there until somebody complaints.